crates/driver/src/infra/solver/mod.rs

```diff
@@ -107,7 +111,16 @@ impl Solver {
             "application/json".parse().unwrap(),
         );
         headers.insert(reqwest::header::ACCEPT, "application/json".parse().unwrap());
-        // TODO(#907) Also add an auth header
+
+        for (key, val) in config.request_headers.iter() {
+            if let Ok(header_name) = HeaderName::try_from(key) {
+                headers.insert(header_name, val.parse().unwrap());
+            }
+            else {
+                panic!("Header Name: {:?} could not be parsed", key);
```

Comment:
I think it would be nice if `new` returned a `Result` and we would `expect` on that result in `run.rs` as this makes it very clear we are panicking during startup (which is fine). If we were ever to refactor our solver creation logic we may end up panicking during operation which is something we want to avoid.

nit: In any case this panic could be replaced by 

```
let header_name = HeaderName::try_from(key).expect("unparseable header");
```

(it's incoherent that `val.parse()` is unwrapped but header parsing isn't.

Reply:
Do you want val parsing unwrapped? I saw that other values you had were unwrapped so I didn't think anything of it
```
headers.insert(
            reqwest::header::CONTENT_TYPE,
            "application/json".parse().unwrap(),
        );
        headers.insert(reqwest::header::ACCEPT, "application/json".parse().unwrap());
```

Reply:
So i had it return a result and then just had the instantiation panic if the result returns an error. Is this what you had in mind? I just used the anyhow crate to propagate those errors. I'm still learning so please be patient but I saw it was used in a bunch of other places so I figured it was fine.